src/bmerge.c

```diff
@@ -50,8 +50,10 @@ SEXP bmerge(SEXP iArg, SEXP xArg, SEXP icolsArg, SEXP xcolsArg, SEXP isorted, SE
   if (LENGTH(icolsArg) > LENGTH(xcolsArg)) error("Internal error: length(icols) [%d] > length(xcols) [%d]", LENGTH(icolsArg), LENGTH(xcolsArg)); // # nocov
   icols = INTEGER(icolsArg);
   xcols = INTEGER(xcolsArg);
-  xN = LENGTH(VECTOR_ELT(x,0));
-  iN = ilen = anslen = LENGTH(VECTOR_ELT(i,0));
+  xN = LENGTH(x) ? LENGTH(VECTOR_ELT(x,0)) : 0;
```

Comment:
Despite the fact that I don't really think a zero-column x will call this function, I add a defense here in case it happens somehow in the future.